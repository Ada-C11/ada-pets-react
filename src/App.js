import React, { Component } from 'react';
import PetList from './components/PetList';
import PetDetails from './components/PetDetails';
import SearchBar from './components/SearchBar';
import NewPetForm from './components/NewPetForm';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

import pets from './data/pets.json';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      petList: pets,
      currentPet: undefined,
    };
  }

  onSelectPet = (petId) => {
    const pet = this.state.petList.find(pet => pet.id === petId)

    this.setState({
      currentPet: pet
    });
  }

  onDeletePet = (petId) => {
    const newPetList = this.state.petList.filter(pet => pet.id !== petId);

    this.setState({
      petList: newPetList
    });
  }

  addPetCallback = (pet) => {
    const petIds = this.state.petList.map(pet => pet.id)

    this.setState({
      petList: [...this.state.petList, {...pet, id: Math.max(...petIds) + 1}]
    });
  }

  searchCallback = (queryString) => {
    this.setState({
      queryString
    });
  }

  render() {
    const { currentPet, petList } = this.state;

    return (
      <main className="App">
        <header className="app-header">
          <h1>Ada Pets</h1>
        </header>
        <section className="search-bar-wrapper">
          <SearchBar searchCallback={this.searchCallback} />
        </section>
        <PetDetails currentPet={currentPet} />
        <section className="pet-list-wrapper">
          <PetList
            pets={petList}
            onSelectPet={this.onSelectPet}
            onDeletePet={this.onDeletePet}
            queryString={this.state.queryString}/>
        </section>
        <section className="new-pet-form-wrapper">
          <NewPetForm addPetCallback={this.addPetCallback} />
        </section>
      </main>
    );
  }
}

export default App;
