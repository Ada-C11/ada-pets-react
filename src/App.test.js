import React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';

describe('<App />', () => {  
  test('that it renders App with shallow rendering', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  test('will match the last snapshot with deep rendering', () => {
    const wrapper = mount(<App />);
    expect(wrapper).toMatchSnapshot();

    // Remove the component from the DOM (save memory and prevent side effects).
    wrapper.unmount();
  });
});