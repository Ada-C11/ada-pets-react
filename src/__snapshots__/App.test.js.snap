// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> that it renders App with shallow rendering 1`] = `
<main
  className="App"
>
  <header
    className="app-header"
  >
    <h1>
      Ada Pets
    </h1>
  </header>
  <section
    className="search-bar-wrapper"
  >
    <SearchBar
      searchCallback={[Function]}
    />
  </section>
  <PetDetails />
  <section
    className="pet-list-wrapper"
  >
    <PetList
      onDeletePet={[Function]}
      onSelectPet={[Function]}
      pets={Array []}
    />
  </section>
  <section
    className="new-pet-form-wrapper"
  >
    <NewPetForm
      addPetCallback={[Function]}
    />
  </section>
</main>
`;

exports[`<App /> will match the last snapshot with deep rendering 1`] = `
<App>
  <main
    className="App"
  >
    <header
      className="app-header"
    >
      <h1>
        Ada Pets
      </h1>
    </header>
    <section
      className="search-bar-wrapper"
    >
      <SearchBar
        searchCallback={[Function]}
      >
        <section>
          <label>
            Search 
            <input
              name="search"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
        </section>
      </SearchBar>
    </section>
    <PetDetails>
      <section
        className="pet-details"
      />
    </PetDetails>
    <section
      className="pet-list-wrapper"
    >
      <PetList
        onDeletePet={[Function]}
        onSelectPet={[Function]}
        pets={Array []}
      >
        <div
          className="card-group"
        />
      </PetList>
    </section>
    <section
      className="new-pet-form-wrapper"
    >
      <NewPetForm
        addPetCallback={[Function]}
      >
        <form
          className="new-pet-form"
          onSubmit={[Function]}
        >
          <h3>
            Add a Pet
          </h3>
          <label>
            Name:
            <input
              name="name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <label>
            Species:
            <select
              name="species"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Please Select a Species
              </option>
              <option
                value="cat"
              >
                Cat
              </option>
              <option
                value="dog"
              >
                Dog
              </option>
              <option
                value="other"
              >
                Other
              </option>
            </select>
          </label>
          <label>
            Location:
            <input
              name="location"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <label>
            About:
            <textarea
              name="about"
              onChange={[Function]}
              value=""
            />
          </label>
          <label>
            Images:
            <input
              name="images"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <input
            className="btn btn-success new-pet-form--submit"
            name="submit"
            type="submit"
            value="Add a Pet"
          />
        </form>
      </NewPetForm>
    </section>
  </main>
</App>
`;
